### PR TITLE
For #25814 - Hide "Show search engines" toggle from search settings when unified search is enabled

### LINKED ISSUE #25814: Remove "Show search engines" toggle when unified search is enabled
Remove "Show search engines" toggle in search settings when unified search is enabled

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/FNXV2-20849)

### PR DESCRIPTION
**Pull request by Alexandru2909**
Hide "Show search engines" toggle from search settings when unified search is enabled
This PR also prevents showing search engine shortcuts after unified search is enabled.

https://user-images.githubusercontent.com/35462038/198674514-b7ede986-ffc5-4c65-bb97-b936ea73fc6f.mp4

### Pull Request checklist

- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [x] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### QA

- [x] **QA Needed**

### To download an APK when reviewing a PR (after all CI tasks finished running):
1. Click on `Checks` at the top of the PR page.
2. Click on the `firefoxci-taskcluster` group on the left to expand all tasks.
3. Click on the `build-debug` task.
4. Click on `View task in Taskcluster` in the new `DETAILS` section.
5. The APK links should be on the right side of the screen, named for each CPU architecture.

### GitHub Automation
Fixes #25814

### CODE REVIEW

File: app/src/test/java/org/mozilla/fenix/search/SearchDialogControllerTest.kt [hunk 1]
```diff
@@ -271,6 +271,29 @@ class SearchDialogControllerTest {
         verify { store.dispatch(SearchFragmentAction.ShowSearchShortcutEnginePicker(true)) }
     }
 
+    @Test
+    fun `do not show search shortcuts when setting enabled AND query empty AND unified search enabled`() {
```
Comment (gabrielluong) [inline app/src/test/java/org/mozilla/fenix/search/SearchDialogControllerTest.kt:275]: Could you change this to use the current GIVEN WHEN THEN format?

**Comment by rocketsroger** [comment]
I think there's a corner case here.  If I turn on show search engine first and then enable unified search, I can't turn it off but still see it with unified search enabled.

**Review by rocketsroger — CHANGES_REQUESTED** [review]
I think there's a corner case here. If I turn on show search engine first and then enable unified search, I can't turn it off but still see it with unified search enabled.

File: app/src/main/java/org/mozilla/fenix/search/SearchFragmentStore.kt [hunk 2]
```diff
@@ -213,7 +213,7 @@ sealed class SearchFragmentAction : Action {
     /**
      * Updates the local `SearchFragmentState` from the global `SearchState` in `BrowserStore`.
      */
-    data class UpdateSearchState(val search: SearchState) : SearchFragmentAction()
+    data class UpdateSearchState(val search: SearchState, val settings: Settings) : SearchFragmentAction()
```
Comment (gabrielluong) [inline app/src/main/java/org/mozilla/fenix/search/SearchFragmentStore.kt:216]: Instead of passing the entire `Settings`, could we just pass in the necessary Boolean value that is needed?

**Comment by Alexandru2909** [comment]
Confirmed with @rocketsroger that we can land this.
